### ISSUE TITLE
Different behaviour DEBUG/RELEASE of m.channels() for an empty matrix

### ISSUE THREAD

**Issue by chacha21** [opencv/opencv#21868]
Windows 7 64b
Visual Studio 2019 16.11.11
OpenCV 4.5.5

The following code queries channels() from a released matrix previously allocated with 3 channels.
```
cv::Mat m = cv::Mat(cv::Size(640, 480), CV_32FC3);
m.release();
std::cout << "m.channels() = " << m.channels() << std::endl;
```

In Debug mode, the output is 1
In Release mode, the output is 3

I don't know the expected behaviour of channels() for an empty matrix. Is it undefined behaviour ?
The context in which I discovered that was by checking `(m.channels()>1)` , expecting false for an empty matrix.

**Comment by HeZean** [comment]
Yeah that's perhaps undefined. It's not meaningful to query the `channels` for a released matrix.

As for the inconsistence, please check below code (`flags`, which contains the number of channels, is assigned to a new value only in debug mode).

https://github.com/opencv/opencv/blob/2b67bc448d2b74103a0202b4bfa0342175df8dfb/modules/core/include/opencv2/core/mat.hpp#L2106-L2112

https://github.com/opencv/opencv/blob/2b67bc448d2b74103a0202b4bfa0342175df8dfb/modules/core/src/matrix.cpp#L547-L565

**Comment by HeZean** [comment]
One possible solution: let the `channels` number be 0 for released matrix, in both debug and release mode.

If that's feasible, I'd like to try fixing it :)
